ros2topic/test/test_echo_pub.py

```diff
@@ -245,3 +245,43 @@ def publish_message():
                     # Cleanup
                     self.node.destroy_timer(publish_timer)
                     self.node.destroy_publisher(publisher)
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_echo_raw(self, launch_service, proc_info, proc_output):
+        topic = '/clitest/topic/echo_raw'
+        publisher = self.node.create_publisher(String, topic, 10)
+        assert publisher
+
+        def publish_message():
+            publisher.publish(String(data='hello'))
+
+        publish_timer = self.node.create_timer(0.5, publish_message)
+
+        try:
+            command_action = ExecuteProcess(
+                cmd=(['ros2', 'topic', 'echo', '--raw', topic, 'std_msgs/msg/String']),
+                additional_env={
+                    'PYTHONUNBUFFERED': '1'
+                },
+                output='screen'
+            )
+            with launch_testing.tools.launch_process(
+                launch_service, command_action, proc_info, proc_output,
+                output_filter=launch_testing_ros.tools.basic_output_filter(
+                    filtered_rmw_implementation=get_rmw_implementation_identifier()
+                )
+            ) as command:
+                # The future won't complete - we will hit the timeout
+                self.executor.spin_until_future_complete(
+                    rclpy.task.Future(), timeout_sec=5
+                )
```

Comment:
@jacobperron this is a bit flaky -- why not waiting for output from the `ros2 topic echo` command?

Reply:
I think this spin is necessary for the test publisher to work (it is triggered by a timer registered above). This is similar logic to the existing test in this file:

https://github.com/ros2/ros2cli/blob/95bfb056df795b6eef8bb73b165db31fcf2d867a/ros2topic/test/test_echo_pub.py#L276-L278

I guess we could additionally wait for output after this spin, which should be more robust. I'll give it a try.

Reply:
See 4472a9e